test/parallel/test-child-process-exec-any-shells-windows.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const cp = require('child_process');
+
+// This test is only relevant on Windows.
+if (!common.isWindows)
+  common.skip('Windows specific test.');
+
+// This test ensures that child_process.exec can work with any shells.
+
+const test = (shell) => {
+  cp.exec('echo foo bar', { shell: shell }, (error, stdout, stderror) => {
+    assert.ok(!error && !stderror);
+    assert.ok(stdout.includes('foo') && stdout.includes('bar'));
+  });
+};
+
+test('cmd');
+test('cmd.exe');
+test('C:\\WINDOWS\\system32\\cmd.exe');
+test('powershell');
+test('C:\\Program Files\\Git\\bin\\bash.exe');
```

Comment:
Is it a requirement to have git bash installed on Windows?

Reply:
Yes, git bash is [one of prerequisites.](https://github.com/nodejs/node/blob/master/BUILDING.md#windows-1)  

>Basic Unix tools required for some tests, Git for Windows includes Git Bash and tools which can be included in the global PATH.

Reply:
bash for windows is not a build requirement, but ATM it's a requirement for running the test suite.

https://github.com/nodejs/node/blob/5384570486d63a75032f47687b5082aeca86faf8/BUILDING.md#L265-L267

However it is not guaranteed to be at `C:\\Program Files\\Git`. This could be done by running `where bash`, and also skipping if it not found.

Reply:
We should not be assuming any paths, e.g. the `SystemDrive` might not be `C:`.

Reply:
Sure, or `execSync`.